src/timeresp.jl

```diff
@@ -118,20 +118,31 @@ function lsim(sys::StateSpace, u::AbstractVecOrMat, t::AbstractVector;
     end
 
     dt = Float64(t[2] - t[1])
-    if !iscontinuous(sys) || method == :zoh
-        if iscontinuous(sys) # Looks strange to check iscontinuous again
+    if !all(x -> x ≈ dt, diff(t))
+        error("time vector t must be uniformly spaced")
+    end
+
+    if iscontinuous(sys)
+        if method == :unspecified
```

Comment:
While `var == :sym` is fine, `var === :sym` is said to be less stressful for the compiler (Ref: Jeff, but I cant remember where). The same goes for comparison to all singleton types, such as `nothing` and `missing`. 
```suggestion
        if method === :unspecified
```

Reply:
We use `==` in quite many other places (and `===` in some), guess consistency would be the main thing (should be an easy find-and-replace). I wouldn't have figured this out by myself, but if it is the way to go then we should change. What do you say @mfalt

Reply:
There is definetely some places where i might have used `==` instead of `===` for symbols and singeltons, if we find them we could fix them.

Reply:
I'll to a find-replace then.